voltixApp/VoltixApp/Views/Send/SendVerifyView.swift

```diff
@@ -1,290 +1,317 @@
-            Text(value)
-                .font(.body13MontserratMedium)
-                .padding(.vertical, 5)
-        }
-    }
-    
-    @ViewBuilder
-    private func LabelTextNumeric(title: String, value: String) -> some View {
-        HStack {
-            Text(title)
-                .font(.body20MenloBold)
-            Spacer()
-            Text(value)
-                .font(.title30MenloUltraLight)
-                .padding(.vertical, 5)
-            Spacer()
-        }
-    }
-    
-    private func estimateGasForEthTransfer() async -> BigInt {
-        
-        do {
-            let estimatedGas = try await web3Service.estimateGasForEthTransaction(senderAddress: tx.fromAddress, recipientAddress: tx.toAddress, value: tx.amountInWei, memo: tx.memo)
-            
-            print("Estimated gas: \(estimatedGas)")
-            
-            return estimatedGas
-        } catch {
-            errorMessage = "Error estimating gas: \(error.localizedDescription)"
-        }
-        return 0
-    }
-    
-    private func estimateGasForERC20Transfer() async -> BigInt {
-        
-        let decimals: Double = Double(tx.token?.tokenInfo.decimals ?? "18") ?? 18
-        
-        let amountInSmallestUnit: Double = tx.amountDecimal * pow(10.0, decimals)
-        
-        let value = BigInt(amountInSmallestUnit)
-        
-        do {
-            let estimatedGas = try await web3Service.estimateGasForERC20Transfer(senderAddress: tx.fromAddress, contractAddress: tx.coin.contractAddress, recipientAddress: tx.toAddress, value: value)
-            
-            print("Estimated gas: \(estimatedGas)")
-            
-            return estimatedGas
-        } catch {
-            errorMessage = "Error estimating gas: \(error.localizedDescription)"
-        }
-        return 0
-    }
+	@Binding var presentationStack: [CurrentScreen]
+	@ObservedObject var tx: SendTransaction
+	@StateObject private var web3Service = Web3Service()
+	@StateObject var unspentOutputsService: UnspentOutputsService = .init()
+	@StateObject var thor: ThorchainService = ThorchainService.shared
+	@StateObject var sol: SolanaService = SolanaService.shared
+	@State private var errorMessage: String = ""
+	@State private var isChecked1 = false
+	@State private var isChecked2 = false
+	@State private var isChecked3 = false
+	
+	private var isValidForm: Bool {
+		return isChecked1 && isChecked2 && isChecked3
+	}
+	
+	private func reloadTransactions() {
+		if tx.coin.chain.name.lowercased() == Chain.Bitcoin.name.lowercased() {
+			if unspentOutputsService.walletData == nil {
+				Task {
+					await unspentOutputsService.fetchUnspentOutputs(for: tx.fromAddress)
+				}
+			}
+		} else if tx.coin.chain.name.lowercased() == Chain.THORChain.name.lowercased() {
+			Task {
+				await thor.fetchAccountNumber(tx.fromAddress)
+			}
+		} else if tx.coin.chain.name.lowercased() == Chain.Solana.name.lowercased() {
+			Task {
+				await sol.getSolanaBalance(account: tx.fromAddress)
```

Comment:
I think you can fire off multiple async at the same time , and then wait for all the result at the end.
FYI: https://www.swiftbysundell.com/articles/swift-concurrency-multiple-tasks-in-parallel/